src/python/pants/backend/docker/subsystems/dockerfile_wrapper_script.py

```diff
@@ -128,16 +145,25 @@ def baseimage_tags(self) -> tuple[str, ...]:
                 stage1 latest
                 out latest
             """
+
+            def _get_tag(image_ref: str) -> str | None:
+                """The image ref is in the form `registry/repo/name[/...][:tag][@digest]` and where
+                `digest` is `sha256:hex value`."""
+                parsed = re.match(_image_ref_regexp, image_ref)
+                if not parsed:
+                    return None
+                tag = parsed.group("tag")
+                if tag:
+                    return tag
+                if not parsed.group("digest"):
+                    return "latest"
+                return None
+
             return tuple(
-                " ".join(
-                    [
-                        stage,
-                        name_parts[-1].rsplit(":", maxsplit=1)[-1]
-                        if ":" in name_parts[-1]
-                        else "latest",
-                    ]
-                )
+                f"{stage} {tag}"
                 for stage, name_parts in self.from_baseimages()
+                for tag in [_get_tag(name_parts[-1])]
```

Comment:
Why use a list here of 1 element?

Reply:
For the `if tag` comprehension condition.

Reply:
Otherwise, there'd potentially be `None` values in the tuple, that we'd have to filter out.